src/OpenTelemetry.Exporter.Jaeger/Implimentation/JaegerUdpBatcher.cs

```diff
@@ -0,0 +1,163 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.Jaeger.Implimentation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Thrift.Protocols;
+
+    public class JaegerUdpBatcher : IJaegerUdpBatcher
+    {
+        private const int DefaultMaxPacketSize = 65000;
+        private readonly int? maxPacketSize;
+        private readonly ITProtocolFactory protocolFactory;
+        private readonly JaegerThriftClientTransport clientTransport;
+        private readonly JaegerThriftClient thriftClient;
+        private readonly Process process;
+        private readonly int processByteSize;
+        private readonly List<JaegerSpan> currentBatch = new List<JaegerSpan>();
+
+        private int batchByteSize;
+
+        private bool disposedValue = false; // To detect redundant calls
+
+        public JaegerUdpBatcher(JaegerExporterOptions options)
+        {
+            this.maxPacketSize = options.MaxPacketSize == 0 ? DefaultMaxPacketSize : options.MaxPacketSize;
+            this.protocolFactory = new TCompactProtocol.Factory();
+            this.clientTransport = new JaegerThriftClientTransport(options.AgentHost, options.AgentPort.Value);
+            this.thriftClient = new JaegerThriftClient(this.protocolFactory.GetProtocol(this.clientTransport));
+            this.process = new Process(options.ServiceName);
+            this.processByteSize = this.GetSize(this.process);
+            this.batchByteSize = this.processByteSize;
+        }
+
+        public async Task<int> AppendAsync(JaegerSpan span, CancellationToken cancellationToken)
+        {
+            int spanSize = this.GetSize(span);
+
+            if (spanSize > this.maxPacketSize)
+            {
+                throw new JaegerExporterException($"ThriftSender received a span that was too large, size = {spanSize}, max = {this.maxPacketSize}", null);
+            }
+
+            this.batchByteSize += spanSize;
+            if (this.batchByteSize <= this.maxPacketSize)
+            {
+                this.currentBatch.Add(span);
+
+                if (this.batchByteSize < this.maxPacketSize)
+                {
+                    return 0;
+                }
+
+                return await this.FlushAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            int n;
+
+            try
+            {
+                n = await this.FlushAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (JaegerExporterException ex)
+            {
+                // +1 for the span not submitted in the buffer above
+                throw new JaegerExporterException(ex.Message, ex);
```

Comment:
Sorry but I don't understand why throwing a new exception of the same type.
Is this really needed?

Reply:
Yeah, this is one of those quick code changes I made while porting their client code. I didn't catch the fact that I was simply throwing the same thing again. Good catch. thanks